ADBench/run-all.ps1

```diff
@@ -169,26 +173,48 @@ function run_command ($indent, $outfile, $timeout, $cmd) {
     $ProcessInfo.Arguments = $args
     $Process = New-Object System.Diagnostics.Process
     $Process.StartInfo = $ProcessInfo
+
     try {
         $Process.Start() | Out-Null
     } catch {
         write-error "Failed to start process $cmd $args"
         throw "failed"
     }
-    $status = $Process.WaitForExit($timeout * 1000)
-    if ($status) {
-        $stdout = $Process.StandardOutput.ReadToEnd().Trim().Replace("`n", "`n${indent}stdout> ")
-        $stderr = $Process.StandardError.ReadToEnd().Trim().Replace("`n", "`n${indent}stderr> ")
-        $allOutput = "${indent}stdout> " + $stdout + "`n${indent}stderr> " + $stderr
-        Write-Host "$allOutput"
-        return [RunCommandStatus]::Finished
-    } else {
-        $Process.Kill()
-        Write-Host "${indent}Killed after $timeout seconds"
-        Store-NonFatalError "Process killed after $timeout seconds`n[$cmd $args]"
-        create_timeout_file $outfile $timeout
-        return [RunCommandStatus]::Timeout
+
+    $OOMCheckingInterval = 500      # milliseconds
+    while (-not $Process.WaitForExit($OOMCheckingInterval)) {
+        $Process.Refresh()
+        if (-not $Process.HasExited) {
+            $mem = 0
+            try {
+                $mem = $Process.PeakPagedMemorySize64
+            } catch {
+                # nothing to do because the process can exit by itself
+            }
+
+            $runTime = ((Get-Date) - $Process.StartTime).Duration().TotalSeconds
+
+            if ($runTime -ge $timeout) {
+                $Process.Kill()
+                Write-Host "${indent}Killed after $timeout seconds"
+                Store-NonFatalError "Process killed after $timeout seconds`n[$cmd $args]"
+                create_timeout_file $outfile $timeout
+                return [RunCommandStatus]::Timeout
+            } elseif ($mem -ge $max_memory_amount_in_bytes) {
+                $Process.Kill()
+                Write-Host "${indent}Killed due to consuming $mem bytes of operating memory"
+                Store-NonFatalError "Process killed due to consuming $mem bytes of operating memory`n[$cmd $args]"
+                return [RunCommandStatus]::OutOfMemory
+            }
+        }
     }
+
+    $stdout = $Process.StandardOutput.ReadToEnd().Trim().Replace("`n", "`n${indent}stdout> ")
+    $stderr = $Process.StandardError.ReadToEnd().Trim().Replace("`n", "`n${indent}stderr> ")
+    $allOutput = "${indent}stdout> " + $stdout + "`n${indent}stderr> " + $stderr
+    Write-Host "$allOutput"
```

Comment:
We should do this even (and especially) when we kill the process due to timeout or OOM.

Reply:
Ok, I will change this. By the way, the last version of the runner didn't do it in the case of timeout as well

Reply:
Hmm.. Indeed. Strange.
Anyway, I think displaying these always is the correct thing to do. And almost trivial to implement.

Reply:
(My opinion on this is that stderr and stdout should be captured, indented and reoutput *immediately* as I noted at https://github.com/awf/ADBench/issues/24#issuecomment-492312312.  It's actually quite annoying to have to wait for a run to finish before seeing its intermediate output.)

Reply:
While I agree in principle, capture-indent-immediate reoutput is not as trivial, so it probably shouldn't a part of this PR.

Reply:
Initial changes are performed